### ISSUE TITLE
Owner bindings will be destroyed

### ISSUE THREAD

**Issue by michellepolte-optimizely** [terraform-google-modules/terraform-google-folders#66]
### TL;DR

After release v4.0 we must change code a bit. But now the concept with folder_admin_roles and all_folder_admins is not properly working anymore.
With v4.x the following roles will be deleted: 
* module.department_folders.google_folder_iam_binding.owners["test-roles/viewer"]
* module.department_folders.google_folder_iam_binding.owners["dev-roles/viewer"]
* module.department_folders.google_folder_iam_binding.owners["prod-roles/viewer"]

### Expected behavior

With new release we want to add additional members with per_folder_admins declaration. We can add separate folders in the per_folder_admins without destroying default roles on not defined folder in that attribute. Like this:

 ```hcl
module "department_folders" {
  source             = "terraform-google-modules/folders/google"
  version            = "~> 4.0"
  parent             = "organizations/${var.organization_id}"
  names              = var.department_names
  set_roles          = true
  folder_admin_roles = ["roles/viewer"]
  all_folder_admins  = ["group:group@test.com"]
  per_folder_admins = {
    test = {
      members = ["group:test@test.com"],
      roles = ["roles/viewer"]
    }
  }
}
```

### Observed behavior

But with that implementation now the permissions for folders dev and prod would be deleted:
```bash
# module.department_folders.google_folder_iam_binding.owners["dev-roles/viewer"] will be destroyed
  # (because key ["dev-roles/viewer"] is not in for_each map)
  - resource "google_folder_iam_binding" "owners" {
     ...
    }
```

### Terraform Configuration

```hcl
department_names = [
  "test",
  "dev",
  "prod"
]

# With v3.x

module "department_folders" {
  source             = "terraform-google-modules/folders/google"
  version            = "~> 4.0"
  parent             = "organizations/${var.organization_id}"
  names              = var.department_names
  set_roles          = true
  folder_admin_roles = ["roles/viewer"]
  all_folder_admins  = ["group:group@test.com"]
}

#With v4.0 we want to add additional members to separate folders with per_folder_admins declaration:

module "department_folders" {
  source             = "terraform-google-modules/folders/google"
  version            = "~> 4.0"
  parent             = "organizations/${var.organization_id}"
  names              = var.department_names
  set_roles          = true
  folder_admin_roles = ["roles/viewer"]
  all_folder_admins  = ["group:group@test.com"]
  per_folder_admins = {
    test = {
      members = ["group:test@test.com"],
      roles = ["roles/viewer"]
    }
  }
}
```

### Terraform Version

```sh
$ terraform --version
Terraform v1.4.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.75.0
+ provider registry.terraform.io/hashicorp/google-beta v4.75.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.5.1
+ provider registry.terraform.io/hashicorp/time v0.9.1
```

### Additional information

It was only as workarround possible to fix that with the following:
```hcl
module "department_folders" {
  source             = "terraform-google-modules/folders/google"
  version            = "~> 4.0"
  parent             = "organizations/${var.organization_id}"
  names              = var.department_names
  set_roles          = true
  folder_admin_roles = ["roles/viewer"]
  all_folder_admins  = ["group:group@test.com"]
  per_folder_admins = {
    test = {
      members = ["group:test@test.com"],
      roles = ["roles/viewer"]
    },
    dev = {
      members = []
    },
    prod = {
      members = []
    }
  }
}
```

**Comment by michellepolte-optimizely** [comment]
Hey, can maybe someone have a look on this issue please? This is not fixed!

**Comment by michellepolte-optimizely** [comment]
This issue is still not fixed.
